marker/v2/builders/document.py

```diff
@@ -15,13 +16,14 @@ def __call__(self, provider: PdfProvider, layout_builder: LayoutBuilder, ocr_bui
         return document
 
     def build_document(self, provider: PdfProvider):
+        PageGroupClass = get_block_cls(PageGroup)
```

Comment:
Why do we pass the potentially overridden class into get_block_cls vs the block type?  (BlockTypes.Page)  That seems more clear to me